src/circuit/gadget/ecc/chip/add.rs

```diff
@@ -190,14 +189,12 @@ pub(super) fn assign_region<C: CurveAffine>(
             .map(|(((x_p, x_q), y_p), y_q)| -> Result<(), Error> {
                 if x_q == x_p {
                     // x_q = x_p ⟹ A
-                    region.assign_advice(|| "set A", a, offset, || Ok(C::Base::one()))?;
+                    region.assign_advice(|| "set A", a, offset, || Ok(F::one()))?;
 
                     // Doubling case, λ = (y_q − y_p) / (x_q − x_p)
-                    if y_p != C::Base::zero() {
-                        let lambda_val = C::Base::from_u64(3)
-                            * x_p
-                            * x_p
-                            * (C::Base::from_u64(2) * y_p).invert().unwrap();
+                    if y_p != F::zero() {
```

Comment:
y_p = 0 should not occur unless we are adding 𝒪+𝒪. It appears we don't assign lambda at all in that case; will this cause a problem?

Reply:
Resolving this comment since `add.rs` has undergone significant fixes.